File: sdk/android/src/main/java/com/dualler/sdk/bridge/Router.kt
package com.dualler.sdk.bridge

import com.dualler.sdk.protocol.BridgeMessage
import com.dualler.sdk.bridge.Handler

class Router {
    fun route(message: BridgeMessage, handlers: Map<String, Handler>): String? {
        val apiName = extractAPIName(message)
        val handler = handlers[apiName]
        return handler?.handle(message)
    }

    private fun extractAPIName(message: BridgeMessage): String? {
        val payload = message.payload as? Map<*, *>
        return payload?.get("api") as? String
    }
}

File: sdk/ios/Sources/DuallerSDK/Router.swift
import Foundation

/// Routes BridgeMessages to registered handlers
public final class Router {
    public func route(_ message: BridgeMessage, handlers: [String: Handler]) -> String? {
        guard let payload = message.payload,
              let apiName = payload["api"] as? String else {
            return nil
        }

        guard let handler = handlers[apiName] else {
            return String(format: "{\"error\":\"No handler for %@\"}", apiName)
        }

        return handler.handle(message)
    }
}
